versions/3.1.0.md

```diff
@@ -2276,11 +2276,16 @@ This object is an extended subset of the [JSON Schema Specification Wright Draft
 For more information about the properties, see [JSON Schema Core](https://tools.ietf.org/html/draft-wright-json-schema-00) and [JSON Schema Validation](https://tools.ietf.org/html/draft-wright-json-schema-validation-00).
 Unless stated otherwise, the property definitions follow the JSON Schema.
 
-##### Properties 
+##### Properties
 
-The following properties are taken directly from the JSON Schema definition and follow the same specifications:
+The OpenAPI Schema Object is a JSON Schema vocabulary which extends JSON Schema Core and Validation vocabularies. As such any keyword available for those vocabularies is by definition available in OpenAPI, and will work the exact same way, including but not limited to:
 
 - title
+- type
+- required
+- allOf
+- oneOf
+- andOf
```

Comment:
Since this Schema object is going to be a superset of JSON Schema Draft-08, should all the `*Of` properties say the following?

"Inline or referenced schema MUST be of a [Schema Object](#schemaObject) and not a standard JSON Schema."

Reply:
Also, looks like a typo in "`andOf`".

Reply:
So... I can add:

```
- properties - Inline or referenced schema MUST be of a [Schema Object](#schemaObject) and not a standard JSON Schema.
- patternProperties - Inline or referenced schema MUST be of a [Schema Object](#schemaObject) and not a standard JSON Schema.
- additionalProperties - Inline or referenced schema MUST be of a [Schema Object](#schemaObject) and not a standard JSON Schema.
- additionalItems - Inline or referenced schema MUST be of a [Schema Object](#schemaObject) and not a standard JSON Schema.
- items - Inline or referenced schema MUST be of a [Schema Object](#schemaObject) and not a standard JSON Schema.
- propertyNames - Inline or referenced schema MUST be of a [Schema Object](#schemaObject) and not a standard JSON Schema.
```

but should I? The point is that it _can_ be a standard JSON Schema object. Before, having a valid JSON Schema object could kill OpenAPI tooling, but now that is not true. The whole point of this thing is to make that not be the case. :)

Reply:
@philsturgeon  Since this is referencing the new draft spec, you could just say that this MUST hold for all applicator keywords, and then you don't have to write it out each time 😁 

And it would apply to extension applicators as well.

Reply:
But it **can** also be a Schema object right? I mean, properties like `externalDocs` **can** be there but they're not part of JSON Schema. Of course, it can be JSON Schema because it's now a subset of the Schema object but it can also be other things that are not JSON Schema but Schema objects.

Reply:
@fmvilas lets avoid saying Schema Object and for the sake of clarity we should prefix each time. 

In v3.0 `schema` accepts an OpenAPI Schema _only_, because giving it valid JSON Schema  would literally break it. 

In v3.1 `schema` accepts an OpenAPI Schema, which is a superset of JSON Schema, so you can give it a pure JSON Schema object, written by somebody who has never heard of OpenAPI, or you can give it a OpenAPI Schema Object (which is JSON Schema + some keywords like xml or externalDocs).

Seeing as it accepts JSON Schema objects, I don't think we should say "You cannot pass it JSON Schema objects" in the docs. Right?

Reply:
Oh yeah, you're right. We should not say you can't. But we should call it for its name and it's the Schema object. Otherwise, what the spec is saying is that you can't use `xml` and `externalDocs` in these properties. Don't you think so?

So we should probably rephrase this to, e.g.:
```
- properties - Inline or referenced schema MUST be of a [Schema Object](#schemaObject) or a standard JSON Schema.
...
```

Reply:
I dont think this wording is disallowing xml, externalDocs, or anything else.

OpenAPI Schema Object is now a superset, which means you can use _more_ keywords in it than exist in JSON Schema, so saying this is completely redundant. 

There is no more "or JSON Schema" because OpenAPI Schema Object _is_ JSON Schema. It is just JSON Schema with a few more keywords available in it. What you're getting at would essentially be this:

> - allOf - We've already said OpenAPI Schema Objects is JSON Schema with a few more keywords in it, and here you can also use those extra keywords, just like anywhere else in `schema`, just in case you forgot
> - anyOf - We've already said OpenAPI Schema Objects is JSON Schema with a few more keywords in it, and here you can also use those extra keywords, just like anywhere else in `schema`, just in case you forgot
> - oneOf - We've already said OpenAPI Schema Objects is JSON Schema with a few more keywords in it, and here you can also use those extra keywords, just like anywhere else in `schema`, just in case you forgot
> - not - We've already said OpenAPI Schema Objects is JSON Schema with a few more keywords in it, and here you can also use those extra keywords, just like anywhere else in `schema`, just in case you forgot
> - items - We've already said OpenAPI Schema Objects is JSON Schema with a few more keywords in it, and here you can also use those extra keywords, just like anywhere else in `schema`, just in case you forgot

😅

Reply:
Maybe think of it in OOP.

In v3.1:

```
class OpenApi extends JsonSchema { }

function schema(JsonSchema $js) { }

schema(new OpenAPI); // yup
schema(new JsonSchema); // yup
```

You would not need to write `function schema(JsonSchema | OpenApi $js) { }` because `OpenApi` is an instance of `JsonSchema`.

Reply:
I may be understanding it wrong but an OpenApi object doesn't have to be a JsonSchema instance. The opposite it's true. All properties of a JsonSchema are present in a OpenApi type but not all the OpenApi properties are present in a JsonSchema type.

Following your example above:

```
class OpenApi extends JsonSchema { }

function schema(JsonSchema $js) { }

schema({"xml": {...}}); // nope! but it must work.
```

Reply:
Snared by my own metaphor! The code might fail depending on the language but ignore the code, that wasn’t the point, and besides JSON Schema allows extra keywords. The main point here is that we are explicitly saying OpenAPI is a JSON Schema Vocabulary, which defines its own extra keywords. It’s not “JSON Schema inspired but different” anymore it’s literally valid meaningful JSON Schema _even with those extra keywords_. An OpenAPI Schema Object is now always and forever a valid JSON Schema instance, there’s just a sad chance that the JSON Schema implementation in question (a validation, or SDK generator, or something else) might not be aware of those extra keywords. It’s still valid JSON Schema. 

There is no need to say “and you can use all the keywords in this bit, and get here, and over here and over here and over here because you can use all the keywords anywhere inside the Schema object. We just need to say once “it’s a valid JSON Schema vocabulary” and if you’re using it on tooling which doesn’t support that OpenAPI vocabulary then most tooling would throw up a warning saying they don’t understand that vocabulary. There’s not a problem here.

Reply:
Ok, I see. Thanks for the explanation. One last question: is the OpenAPI schema going to allow extra properties as JSON Schema does? I mean, without counting the extensions. For instance:

```yaml
type: string
whatever: "whatever value"
```

Would this be a valid OpenAPI schema?

Reply:
Great question! I'd like to say yes, but ... this is a bigger topic for elsewhere. Lets jump up to main thread and resolve this bit.